test/jdk/java/awt/TrayIcon/DisposeInActionEventTest/DisposeInActionEventTest.java

```diff
@@ -21,76 +21,73 @@
-    private boolean traySupported;
 
-    private SystemTray systemTray;
-    private TrayIcon trayIcon;
+/*
+ * @test
+ * @bug 6299866
+ * @summary Tests that no NPE is thrown when the tray icon is disposed from the
+ * handler of action event caused by clicking on this icon.
+ * @library ../../regtesthelpers /test/lib
+ * @build Sysout jtreg.SkippedException
+ * @run main/manual DisposeInActionEventTest
+ */
 
-    public void init() {
-        this.setLayout(new BorderLayout());
+public class DisposeInActionEventTest {
+    private static JTextArea textArea;
 
-        String[] instructions;
-        traySupported = SystemTray.isSupported();
-        if (!traySupported) {
-            instructions = new String[]{
-                    "The test cannot be run because SystemTray is not supported.",
-                    "Simply press PASS button."
-            };
-        } else {
-            String clickInstruction;
-            if (Platform.isOSX()) {
-                clickInstruction = "right";
-            } else {
-                clickInstruction = "left";
-            }
-            instructions = new String[]{
-                    "When the test starts, it adds the icon to the tray aread. If you",
-                    "  don't see a tray icon, please, make sure that the tray area",
-                    "  (also called Taskbar Status Area on MS Windows, Notification",
-                    "  Area on Gnome or System Tray on KDE) is visible.",
-                    "Double-click with " + clickInstruction + " button on the tray icon to trigger the",
-                    "  action event. Brief information about action events is printed",
-                    "  below. After each action event the tray icon is removed from",
-                    "  the tray and then added back in a second.",
-                    "The test performs some automatic checks when removing the icon. If",
-                    "  something is wrong the corresponding message is displayed below.",
-                    "  Repeat double-clicks several times. If no 'Test FAILED' messages",
-                    "  are printed, press PASS button else FAIL button."
-            };
+    public static void main(String[] args) throws Exception {
+        if (!SystemTray.isSupported()) {
+            throw new jtreg.SkippedException("The test cannot be run because " +
+                    "SystemTray is not supported.");
         }
-        Sysout.createDialogWithInstructions(instructions);
+        String clickInstruction = (Platform.isOSX()) ? "Right-click" : "Double-left click";
+
+        String instructions = "When the test starts, it adds the icon to the tray area. If you\n" +
+                       "  don't see a tray icon, please, make sure that the tray area\n" +
+                       "  (also called Taskbar Status Area on MS Windows, Notification\n" +
+                       "  Area on Gnome or System Tray on KDE) is visible.\n" +
```

Comment:
Instructions look more clear with the addition of newlines as below.

```suggestion
                       "  Area on Gnome or System Tray on KDE) is visible.\n\n" +
```